### ISSUE TITLE
Conflicting hr.leaves refuse

### ISSUE THREAD

**Issue by aaly-dp** [odoo/odoo#67335]
Impacted versions:
Odoo 14

Steps to reproduce:
 1. Create a hr.leave for the whole company with an hr.leave.type "Public Holiday" for half a day (like 4hours) and approve it, so every employee has half a day off for the 01.03.2021.
 2. Create another hr.leave for 3 days from 01.03.2021 - 03.03.2021 with the hr.leave.type "Vacation" for the whole company and approve it, so every employee has time off from 01.03.2021 - 03.03.2021.
Current/Expected behavior:
Currently the Odoo will refuse the 1st hr.leave(Public holiday), because of the conflict with the second leave. The problem with this logic is that there are countries like Austria, where you get half a day off for 24.12.2021 and half a day off for 31.12.2021, and many companies go on an extended vacation from the 24.12.2021 - 02.02.2022, because almost the whole country does the same thing.
Which leads to the current problem that Odoo refuses the "Public Holiday", which means you lose half a day of vacation, because you can't have overlapping timeoffs like that. 
If you do those steps for an individual employee than it works, but it can't be expected to do that for multiple employees.
Video/Screenshot link (optional):
![screenshot_1](https://user-images.githubusercontent.com/32839388/110108147-1e3b9400-7dac-11eb-8bfa-f94eac4d8e23.png)
Support ticket number submitted via odoo.com/help (optional):

**Comment by danielhasan1** [comment]
Hello, Can I contribute to any of these issues where a solution is not already submitted?

**Comment by Yenthe666** [comment]
@danielhasan1 yes. If you have a solution you're free to make a pull request to try and fix an issue.
